lms-backend/src/main/scala/scalan/compilation/lms/common/LstOps.scala

```diff
@@ -90,7 +90,7 @@ trait ScalaGenLstOps extends ScalaNestedCodegen {
   override def emitNode(sym: Sym[Any], rhs: Def[Any]) = rhs match {
     case ListReplicate(len, x) => emitValDef(sym, src"List.fill($len)($x)")
     case ListCreateAndFill(length, elem) => emitValDef(sym, src"List.fill(${quote(length)})(${quote(elem)})")
-    case ListRangeFrom0Lms(length) => emitValDef(sym, src"List((for(i <- 0 to ${quote(length)}) yield i ) : _*)")
+    case ListRangeFrom0Lms(length) => emitValDef(sym, src"List((for(i <- 0 until ${quote(length)}) yield i ) : _*)")
```

Comment:
+ First, we can construct range `[x, y)` `y` is not included, by `List.range(x, y)`. I suspect old logic was made on purpose. @alexeyr is it possible that some code could rely on the old implementation?
+ Second, I see the issue with [class ListRangeFrom0Lms](https://github.com/scalan/scalan/pull/87/files#diff-b1f86dda9efe0fe4b9e5123fd29ef3f3L22). It should extend `Def[List[Int]]`, not `Def[Int]` IMO. I wander if this method even works...
+ Third, since behavior was changed, I believe there should be some test showing it.